web/api/v1/api.go

```diff
@@ -1217,6 +1228,10 @@ func (api *API) metricMetadata(r *http.Request) apiFuncResult {
 				m := metadata{Type: md.Type, Help: md.Help, Unit: md.Unit}
 				ms, ok := metrics[md.Metric]
 
+				if len(metrics[md.Metric]) >= limitPerMetric {
+					break
```

Comment:
Is `break` correct here? Wouldn't you be skipping through all subsequent targets?

You may be missing a unit test where the request inputs a metric.

Reply:
Not sure what you mean by targets here 🤔, on this API we're retrieving metadata from metrics that are already stored and nothing related to scrape time.

Did you mean that we might be skipping metadata from subsequent metrics?

Reply:
If so, that was my initial line of thought too! But after debugging it seems like we're already storing multiple metadata from the same metric?

Anyway, the only test case implemented so far(I'll add more) shows that we're not skipping subsequent metrics and only limiting metadata

Reply:
By targets, I mean variable `t` is of type `scrape.Target`, which is what internally stores the metadata store. I'm not sure how the metadata store works, but if each target stores different metadata, then your code is skipping all metadata after you reach the limit. This would skip metadata from subsequent metrics.

If you had a unit test with different metadata in a single target, we'd be able to verify this easily!

Reply:
Yep, you're correct! It should be `continue` instead of `break`, test confirmed

Reply:
I'll add more tests later :)